asv_bench/benchmarks/algorithms.py

```diff
@@ -28,23 +28,20 @@ class Factorize:
             "datetime64[ns, tz]",
             "Int64",
             "boolean",
-            "string_arrow",
+            "string[pyarrow]",
```

Comment:
i guess should be consistent about using `string[python]` to be more explicit (rather than 'string'). i think its worth it in benchmarks for example. (and you do it on other benchmarks)

Reply:
the "string" used here denotes an object Index. These are not dtypes, but dictionary keys. There is no benchmark for StringArray.factorize

The last 3 are benchmarking arrays, all the others are Indexes

Reply:
I could maybe call it ArrowStringArray and rename the others for clarity. (does that affect the asv history?)

Reply:
not sure about asv history, but nbd.

Reply:
`string` -> `object` in https://github.com/pandas-dev/pandas/pull/39908/commits/3e26baa7c29402010538db70be59c59681fe9889